### PR TITLE
TMEDIA-595 Add fetching next data set to collection API

### PR DESCRIPTION
**Pull request by matthewroach**
## Description

Switch the collection API to use fetch to allow fetching the next "set" of results from a collection due to the limits of the collection API being set to maximum of 20

## Jira Ticket
- [TMEDIA-595](https://arcpublishing.atlassian.net/browse/TMEDIA-595)

## Acceptance Criteria
_copy from ticket_

## Test Steps

1. Checkout this branch `git checkout TMEDIA-595-collection-fetch-next`
2. Run `npm i` from the root of the blocks repo
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/collections-content-source-block,@wpmedia/results-list-block`
3. Using the results list block with collections set up, see more button is now working as expected when using any size value upto 20

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

**Review by JackHowa — APPROVED** [review]
Working as expected. Makes sense

before

blank screen 

after 

- shows items 
Screen Shot 2022-01-05 at 10 47 35

File: blocks/collections-content-source-block/package.json [hunk 1]
```diff
@@ -26,5 +26,8 @@
     "@wpmedia/engine-theme-sdk": "*",
     "@wpmedia/resizer-image-block": "*"
   },
-  "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95"
+  "gitHead": "83fb8d6685958d85ddbfcfbf01d9a5c864cd6c95",
+  "dependencies": {
+    "request-promise-native": "^1.0.9"
```
Comment (ecarlisle) [inline blocks/collections-content-source-block/package.json]: Not a blocker, but more so a general question - Have we considered using another HTTP client that handles promises? `request-promise-native` is deprecated and libraries like `axios` are just as capable and smaller.
https://www.npmtrends.com/axios-vs-request-promise-native

Comment (matthewroach) [reply blocks/collections-content-source-block/package.json]: @ecarlisle I've switched to use `axios` now, this was my bad - I use the library that was referenced in the ALC documentation for content sources

File: blocks/collections-content-source-block/sources/content-api-collections.js [hunk 2]
```diff
@@ -1,31 +1,64 @@
...
   } = key;
   let updatedSize = size;
   // Max collection size is 20
   // See: https://redirector.arcpublishing.com/alc/docs/swagger/?url=./arc-products/content-api.json
   // Want to ensure size is capped at 20 to prevent an error.
   if (updatedSize && updatedSize > 9) updatedSize = size < 20 ? size : 20;
-  return `content/v4/collections?${_id ? `_id=${_id}` : `content_alias=${contentAlias}`}${site ? `&website=${site}` : ''}${from ? `&from=${from}` : ''}${size ? `&size=${updatedSize}` : ''}&published=true`;
+
+  return axios({
+    url: `${CONTENT_BASE}content/v4/collections?${_id ? `_id=${_id}` : `content_alias=${contentAlias}`}${site ? `&website=${site}` : ''}${from ? `&from=${from}` : ''}${size ? `&size=${updatedSize}` : ''}&published=true`,
```
Comment (ecarlisle) [inline blocks/collections-content-source-block/sources/content-api-collections.js]: Do we need a `/` between `${CONTENT_BASE}` and `content/`?
I don't currently have a trailing `/` in the `CONTENT_BASE` in my `.env`. so I'm seeing the following:
image
I'm comparing this to `content-api-source-block` where we have a `/` preceding `content`.
https://github.com/WPMedia/arc-themes-blocks/blob/canary/blocks/content-api-source-block/sources/content-api.js#L11

**Review by ecarlisle — CHANGES_REQUESTED** [review]

File: blocks/collections-content-source-block/sources/content-api-collections.js (hunk 2 above)
Comment (ecarlisle) [reply blocks/collections-content-source-block/sources/content-api-collections.js]: (I'll change my `.env` to include a trailing `/` in the `CONTENT_BASE` in the meanwhile to continue testing)

Comment (matthewroach) [reply blocks/collections-content-source-block/sources/content-api-collections.js]: That would depend on how the `CONTENT_BASE` is set up in a non local environment. I'll check with the Engine team

Comment (matthewroach) [reply blocks/collections-content-source-block/sources/content-api-collections.js]: Update the code to add the slash in, confirmed from Engine that the generated `CONTENT_BASE` will not contain it

**Comment by ecarlisle** [comment]
When I changed my `.env` to have a trailing `/` on the `CONTENT_BASE`, the block operates as expected. However, I needed to ensure that the collection belonged to the site I was previewing in PB Editor.  I used the Results List when previewing a page for The Sun and configured it using a collection id configured for The Gazette. I could see a 404 response in the Docker log, but there was no "Not Found" feedback in the PB Editor Preview. 
This could potentially be confusing UX for an Editor. We may want to consider this for a future ticket.

**Review by ecarlisle — APPROVED** [review]

**Pull request merged**